tests_integration/test_sql.py

```diff
@@ -36,3 +39,43 @@ def _round_values(results):
 
 def _round_values_for_multi_queries(results):
     return [_round_values(r) for r in results]
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "input_, query",
+    [
+        ("select foo;", "select foo;"),
+        ("select 1; select foo; select 2;", "select foo;"),
+    ],
+)
+def test_execute_adds_failing_query_to_output(input_, query, runner):
+    result = runner.invoke_with_connection_json(
+        ["sql", "-q", input_], catch_exceptions=True
+    )
+    assert query in result.output
+
+
+@pytest.mark.integration
+@mock.patch("snowcli.output.printing._get_table")
+@mock.patch("snowcli.output.printing.Live")
+def test_queries_are_streamed_to_output(
+    _, mock_get_table, runner, capsys, test_root_path
+):
+    runner.invoke_with_connection(
+        [
+            "sql",
+            "-q",
+            "select CURRENT_TIME(); select system$wait(10); select CURRENT_TIME();",
+        ],
+    )
+
+    add_row = mock_get_table().add_row
+    results = [p.args[0] for p in add_row.mock_calls]
+    assert len(results) == 3
+    start, wait, end = results
+    assert wait == "waited 10 seconds"
+    start_ts = datetime.strptime(start[:8], "%H:%M:%S")
+    end_ts = datetime.strptime(end[:8], "%H:%M:%S")
+    duration = (end_ts - start_ts).total_seconds()
```

Comment:
I'm not sure if this test is really able to verify that results are streamed. 
Checking duration between timestamps generated on the server side won't check that the results are really printed on client side in the right time.